pkg/webhook/seed/validation.go

```diff
@@ -182,6 +182,10 @@ func (v *validator) validate(ctx context.Context, obj runtime.Object, isDelete b
 	}
 
 	if subject.Spec.EtcdBackupRestore != nil {
+		if !v.features.Enabled(features.EtcdLauncher) {
+			return errors.New("EtcdLauncher feature must be enabled for etcd backup and restore")
+		}
```

Comment:
I think Clusters can have `etcd-launcher` enabled even without it being enabled for the seed. So in that case this wont work properly.

Why not just check somewhere here, and skip the restore if the cluster doesnt have `etcd-launcher` enabled?
https://github.com/kubermatic/kubermatic/blob/master/pkg/controller/seed-controller-manager/etcdrestore/etcd_restore_controller.go#L120-L145

Reply:
> Why not just check somewhere here, and skip the restore .... 

This (etcd backup/restore) is a feature that depends upon other feature(etcd-launcher) being enabled. So we decided to

Reply:
> I think Clusters can have etcd-launcher enabled even without it being enabled for the seed. So in that case this wont work properly.

I see. Hmm... could we make it not possible? or it is by design?

Reply:
Individual clusters having the `etcdLauncher` feature flag enabled while the Seed or KubermaticConfiguration does not is by design. You can, for example, [disable etcd-launcher on a global level](https://docs.kubermatic.com/kubermatic/v2.20/cheat_sheets/etcd/etcd-launcher/#disabling-etcd-launcher), but it cannot be rolled back on existing clusters, so that would be a scenario where this way of validating it might become problematic.

Reply:
Gotcha. Thanks! 

So we have to go back to the first PR that I pushed on this issue :stuck_out_tongue:

Reply:
@lsviben @embik Ptal. https://github.com/kubermatic/kubermatic/pull/9633